demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/Navigation.kt

```diff
@@ -107,16 +107,27 @@ private fun NavHostController.popBackStackJustOnce(currentRouteToBePopped: Strin
     popBackStack()
 }
 
-// Note: this requires API level 23 (current min is 21 for the Demo-app). We should address this requirement, or set
-// our Demo-app min to 23
-@TargetApi(Build.VERSION_CODES.M)
 fun copyToClipboard(context: Context, tag: String, textToCopy: String) {
-    val clipboardManager = context.getSystemService(ClipboardManager::class.java)
+    val clipboardManager = if (AndroidApiVersion.isAtLeastM) {
+        context.getSystemService(ClipboardManager::class.java)
+    } else {
+        context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+    }
+
     val data = ClipData.newPlainText(
         tag,
         textToCopy
     )
     clipboardManager.setPrimaryClip(data)
+
+    // Notify users with Toast only on Android level 32 and lower, as 33 notifies users by its own system way
```

Comment:
Let's avoid introducing Toast.  Although this is just a demo UI, Toast has problems such as being disabled if notifications for the app are disabled.  Snackbar would provide a better example for other developers to copy.

Would you prefer to just leave out the Toast entirely for now, or rework as a snackbar?

Reply:
Yep, good point. I vote for reworking it now to have this done.